lib/full-site-editing/template-parts.php

```diff
@@ -60,6 +60,49 @@ function gutenberg_register_template_part_post_type() {
 }
 add_action( 'init', 'gutenberg_register_template_part_post_type' );
 
+/**
+ * Registers the 'wp_template_section' taxonomy.
+ */
+function gutenberg_register_wp_template_section_taxonomy() {
+	if ( ! gutenberg_is_fse_theme() ) {
+		return;
+	}
+
+	register_taxonomy(
+		'wp_template_section',
+		array( 'wp_template_part' ),
+		array(
+			'public'            => false,
+			'hierarchical'      => false,
+			'labels'            => array(
+				'name'          => __( 'Sections', 'gutenberg' ),
+				'singular_name' => __( 'Section', 'gutenberg' ),
+			),
+			'query_var'         => false,
+			'rewrite'           => false,
+			'show_ui'           => false,
+			'_builtin'          => true,
+			'show_in_nav_menus' => false,
+			'show_in_rest'      => false,
+		)
+	);
+}
+add_action( 'init', 'gutenberg_register_wp_template_section_taxonomy' );
+
+// Definte constants for supported WP_TEMPLATE_SECTION taxonomy.
+if ( ! defined( 'WP_TEMPLATE_SECTION_HEADER' ) ) {
+	define( 'WP_TEMPLATE_SECTION_HEADER', 'header' );
+}
+if ( ! defined( 'WP_TEMPLATE_SECTION_FOOTER' ) ) {
+	define( 'WP_TEMPLATE_SECTION_FOOTER', 'footer' );
+}
+if ( ! defined( 'WP_TEMPLATE_SECTION_SIDEBAR' ) ) {
+	define( 'WP_TEMPLATE_SECTION_SIDEBAR', 'sidebar' );
+}
+if ( ! defined( 'WP_TEMPLATE_SECTION_OTHER' ) ) {
+	define( 'WP_TEMPLATE_SECTION_OTHER', 'other' );
```

Comment:
Not super sure "other" should be defined. Should we instead go with just plain uncategorized?

Reply:
Maybe that makes more sense.  My initial thought was that TPs that dont have a section term defined yet (such as the theme not supplying the term for their TPs, or the cpt being created elsewhere where the section is not specified) would not have a section set and thus correspond to being uncategorized.  While TPs that are created with a section term that is unsupported (like 'diagonal sidepanel' as a goofy example) would be categorized as 'other'.  So 'other' being different than uncategorized as: "we tried to categorize it, but it didnt make sense."

Do you think it makes more sense for both cases to just fall under uncategorized together?  Maybe thats a more simple way to go about it... im definitely open to doing it that way instead.

Reply:
Updated so both cases just go with uncategorized.

Reply:
Thanks, that seems like a better start. It's always easier to add more terms / concepts later than to remove them.